### PR TITLE
[DOCS] Move anomaly detection job resource definitions into APIs

### PR DESCRIPTION
**Pull request by lcawl**
This PR adds tagged sections that contain descriptions of the anomaly detection job resource properties in https://github.com/elastic/elasticsearch/blob/master/docs/reference/ml/ml-shared.asciidoc. It removes the job resource definitions page (https://www.elastic.co/guide/en/elasticsearch/reference/master/ml-job-resource.html) and uses the shared tagged sections to put that information directly into the appropriate APIs. 

NOTE: As a result of these changes, users no longer need to flip between the job resource page and the API page to find all the necessary details. However, from a maintenance stand-point, it means that when parameters or properties are added or removed, each of the affected APIs' pages must be updated appropriately.

Subsequent PRs will be opened to (1) add missing indicators of which properties are required or optional in the create job API, (2) minimize some of the overly long property descriptions, and (3) merge some of the varied definitions for "job_id", since it seems unlikely so many variations are needed.

Preview:
http://elasticsearch_49700.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/ml-put-job.html
http://elasticsearch_49700.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/ml-get-job.html
http://elasticsearch_49700.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/ml-update-job.html
...

### CODE REVIEW

**Review by edsavage — COMMENTED** [review]
Looks like the CI build failed due to...

```
WARNING: ml/anomaly-detection/apis/put-job.asciidoc: line 36: tag 'job-id-def' not found in include file: /tmp/docsbuild/cyKWLdPar0/elasticsearch/docs/reference/rest-api/common-parms.asciidoc
```

**Comment by edsavage** [comment]
CI failed due to

> Invalid json in reference/ml/anomaly-detection/apis/get-job.asciidoc[154:207](console-result)// TESTRESPONSE. The error is:
23:18:50   expecting '}' or ',' but got current char '.' with an int value of 46

This is due to some dodgy JSON introduced in commit 9c997cf

**Comment by lcawl** [comment]
@elasticmachine elasticseach-ci/docs-check

**Comment by lcawl** [comment]
@elasticmachine run elasticsearch-ci/docs-check

File: docs/reference/ml/ml-shared.asciidoc [hunk 1]
```diff
@@ -1,12 +1,378 @@
...
+`time_format`:::
+(string)
+include::{docdir}/ml/ml-shared.asciidoc[tag=time-format]
+--
+end::data-description[]
+
 tag::dependent_variable[]
-`dependent_variable`::
-(Required, string) Defines which field of the document is to be predicted. 
+Defines which field of the document is to be predicted.
```
Comment (szabosteve) [inline docs/reference/ml/ml-shared.asciidoc]: I'm wondering if we can put it back until I start to work on the DFA related parameters? I'd like to be consistent, too, but then we need to delete the name and the info between parentheses in the case of dependent_variable, gamma, eta, lambda, feature_bag_fraction, maximum_number_trees, and so on, then open a PR that adds the removed info back to the API docs. Finally, merge and backport the two PRs (this one and the one that put the titles back to the API docs) in sync.
The other option is to leave it as-is for now and I'll remove the titles in this file and add them back in the API docs in one PR later. WDYT? If my train of thought is obscure and I explained my concern poorly, please let me know.

```suggestion
`dependent_variable`::
(Required, string) Defines which field of the document is to be predicted. 
```

File: docs/reference/ml/ml-shared.asciidoc [hunk 2]
```diff
@@ -46,25 +490,198 @@ relevant relationships between the features and the {depvar}. The smaller this
...
+If your `elasticsearch.yml` file contains an `xpack.ml.max_model_memory_limit`
+setting, an error occurs when you try to create jobs that have
+`model_memory_limit` values greater than that setting. For more information,
+see <<ml-settings>>.
+--
+end::model-memory-limit[]
+
+tag::model-plot-config[]
+This advanced configuration option stores model information along with the
+results. It provides a more detailed view into anomaly detection.
```
Comment (szabosteve) [inline docs/reference/ml/ml-shared.asciidoc]:
```suggestion
results. It provides a more detailed view into {anomaly-detect}.
```

**Review by szabosteve — APPROVED** [review]
LGTM!
I left a few comments and suggestions, nothing blocker though. Thank you for this enormous work!

File: docs/reference/ml/ml-shared.asciidoc (hunk 1 above)
Comment (lcawl) [reply docs/reference/ml/ml-shared.asciidoc]: Agreed, there was no reason for me to change this, since it isn't used by any of the APIs I updated.

**Pull request merged**